### PR TITLE
DTSPO-14865-replace-kured

### PR DESCRIPTION
**Pull request by JoshuaGlassKainos**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-14865

### Change description ###
added node_os_channel_upgrade to enable when env is sbox

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Comment by JoshuaGlassKainos** [comment]
https://github.com/hmcts/aks-sds-deploy/blob/eb15d31d007eb03f283f8b2963f33f4c9ea1054b/components/aks/aks.tf#L141
`enable_node_os_channel_upgrade_nodeimage = var.enable_node_os_channel_upgrade_nodeimage`

this flag is permanently on if env is sandbox, it can be abstracted further by replacing it with a tfvars like the following `enable_node_os_channel_upgrade_nodeimage = true` in tfvars

File: components/aks/aks.tf [hunk 1]
```diff
@@ -136,6 +136,9 @@ module "kubernetes" {
   aks_auto_shutdown_principal_id = data.azuread_service_principal.aks_auto_shutdown.object_id
 
   enable_automatic_channel_upgrade_patch = var.enable_automatic_channel_upgrade_patch
+
+  # optional flag, if env is sbox then set flag to true
```
Comment (reespozzi) [inline components/aks/aks.tf:140]: Would remove this comment, pretty clear in the code I think

**Review by reespozzi — CHANGES_REQUESTED** [review]

File: components/aks/aks.tf (hunk 1 above)
Comment (thomast1906) [reply components/aks/aks.tf:140]: +1 most terraform should be made readable without the need for comments

**Comment by reespozzi** [comment]
Does the feature need adding to sbox aks before merging this or?

**Comment by JoshuaGlassKainos** [comment]
> Does the feature need adding to sbox aks before merging this or?

Will need to look into this further, but I think `Microsoft.ContainerService/NodeOsUpgradeChannelPreview` will need to enabled on sbox cluster for the `node_os_channel_upgrade` to take effect

**Comment by JoshuaGlassKainos** [comment]
Sbox will need the following commands to be ran to ensure the preview feature `Microsoft.ContainerService/NodeOsUpgradeChannelPreview` is active on that cluster.
[link to ref](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image#register-nodeosupgradechannelpreview)

Enable preview feature on the cluster with the following 
`az feature register --namespace "Microsoft.ContainerService" --name "NodeOsUpgradeChannelPreview"`

Register preview feature with the following
`az provider register --namespace Microsoft.ContainerService`

**Review by JoshuaGlassKainos — COMMENTED** [review]
comments removed and fat file

**Review by reespozzi — APPROVED** [review]

**Comment by reespozzi** [comment]
Could you FMT again please? Not sure it's been run

**Pull request merged**